src/signatures/mod.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright 2019-2020 Centrality Investments Limited
+// Few code copied from wasm/wasm-crypto, license header to be updated
+// https://github.com/polkadot-js/wasm/blob/master/packages/wasm-crypto/src/ed25519.rs
+
+use crate::error::VerifyError;
+use core::convert::TryFrom;
+use ed25519_dalek::{
+    Keypair as Ed25519Keypair, PublicKey as Ed25519PublicKey, SecretKey as Ed25519SecretKey,
+    Signature as Ed25519Signature,
+};
+use schnorrkel::{
+    derive::{ChainCode, Derivation, CHAIN_CODE_LENGTH},
+    signing_context, ExpansionMode, Keypair as Sr25519Keypair,
+    MiniSecretKey as Sr25519MiniSecretKey, PublicKey as Sr25519PublicKey,
+    SecretKey as Sr25519SecretKey, Signature as Sr25519Signature,
+};
+pub const CONTEXT_ID: &[u8] = b"substrate";
+
+/// copied from wasm/wasm-crypto
+fn create_from_pair(pair: &[u8]) -> Ed25519Keypair {
+    match Ed25519Keypair::from_bytes(pair) {
+        Ok(pair) => return pair,
+        Err(_) => panic!("Provided pair is invalid."),
+    }
+}
+
+/// copied from wasm/wasm-crypto
+fn create_ed25519_keypair_from_keys(pubkey: &[u8], seckey: &[u8]) -> Ed25519Keypair {
+    let mut pair = vec![];
+
+    pair.extend_from_slice(seckey);
+    pair.extend_from_slice(pubkey);
+
+    create_from_pair(&pair)
+}
+
+fn create_ed25519_public_key(pubkey: &[u8]) -> Ed25519PublicKey {
+    match Ed25519PublicKey::from_bytes(pubkey) {
+        Ok(pubkey) => return pubkey,
+        Err(_) => panic!("Provided public key is invalid."),
+    }
+}
+
+/// Sign an ed25519 signature
+pub fn sign_ed25519(public_key: &[u8], private_key: &[u8], payload: &[u8]) -> Vec<u8> {
+    let keypair = create_ed25519_keypair_from_keys(public_key, private_key);
+    keypair.sign(payload).to_bytes().to_vec()
+}
+
+/// Verify an ed25519 signature
+pub fn verify_ed25519(signature: &[u8], payload: &[u8], public_key: &[u8]) -> bool {
+    let signature = match Ed25519Signature::from_bytes(signature) {
+        Ok(signature) => signature,
+        Err(_) => return false,
+    };
+
+    create_ed25519_public_key(public_key)
+        .verify(payload, &signature)
+        .is_ok()
+}
+
+/// Sign an sr25519 signature
+pub fn sign_sr25519(public_key: &[u8], private_key: &[u8], payload: &[u8]) -> Vec<u8> {
```

Comment:
@holygits I relocated new signing/verifying functions into a separated module, and will merge it with the existing verifying method if necessary before final review/PR merge, is it ok?